src/Polly.Core/CircuitBreaker/BrokenCircuitException.cs

```diff
@@ -76,6 +76,31 @@ public BrokenCircuitException(string message, TimeSpan retryAfter, Exception inn
     protected BrokenCircuitException(SerializationInfo info, StreamingContext context)
         : base(info, context)
     {
+        Guard.NotNull(info);
+
+        // https://github.com/dotnet/runtime/issues/42460
+        SerializationInfoEnumerator enumerator = info.GetEnumerator();
+        while (enumerator.MoveNext())
+        {
+            SerializationEntry entry = enumerator.Current;
+            if (string.Equals(entry.Name, "RetryAfter", StringComparison.Ordinal))
+            {
+                var ticks = (long)entry.Value;
+                RetryAfter = new TimeSpan(ticks);
```

Comment:
Can break here as we've found the only member we're looking for, though it does make me wonder why we're doing this approach at all rather than just always setting `RetryAfter` into the `SerializationInfo` and then read it out with `GetValue()`? Do we need to make this ultra-definsive about people passing in a `SerializationInfo` that wasn't populated by us?

Reply:
Because the property is nullable and the current implementation does not add the value when it's null. Also I think we'd better defend against users who deserialize from persisted data generated by previous versions. 

As to why not `break`ing, I think it avoids future problems where we add more properties later without revisiting and removing the `break`. But that's opinion.

Reply:
Ok, in that case let's just add the break. The need to remove it will become apparent in testing if we ever add any additional properties.